lib/services/music_player_background_task.dart

```diff
@@ -1082,20 +1088,59 @@ class MusicPlayerBackgroundTask extends BaseAudioHandler {
   double get volume => _player.volume;
   bool get paused => !_player.playing;
   Duration get playbackPosition => _player.position;
+
+  void onQueueServiceAvailable() {
+    // Moved here because currentTrackMetadataProvider depends on queueService
+    // If metadataProvider is sloooooow, this allows it to catch up
+    _providerContainer.listen(currentTrackMetadataProvider, (previous, next) {
+      if (FinampSettingsHelper.finampSettings.volumeNormalizationMode != VolumeNormalizationMode.albumBased &&
+          FinampSettingsHelper.finampSettings.volumeNormalizationMode != VolumeNormalizationMode.hybrid) {
+        return;
+      }
+      if (previous?.valueOrNull?.parentNormalizationGain != next.valueOrNull?.parentNormalizationGain) {
+        _applyVolumeNormalization(mediaItem.valueOrNull);
+      }
+    });
+  }
 }
 
-double? getEffectiveGainChange(MediaItem currentTrack, jellyfin_models.BaseItemDto? item) {
+extension SequentalTracks on FinampQueueInfo {
+  bool isSequental() {
+    final currentTrackAlbum = currentTrack?.baseItem?.parentId;
+    if (currentTrackAlbum == null) return false;
+    final previousTrackAlbum = previousTracks.lastOrNull?.baseItem?.parentId;
+    final nextTrackAlbum = (nextUp.firstOrNull ?? queue.firstOrNull)?.baseItem?.parentId;
+
+    return previousTrackAlbum == currentTrackAlbum || currentTrackAlbum == nextTrackAlbum;
+  }
+}
```

Comment:
why is this here instead of in `queue_service`? 🤔

Reply:
The same as albumProvider (which should be `itemProvider`) - tunnel vision :-)

I actually want it to be private. The only reason I returned this extension back (I inlined it once) is to neatly use it after `when` and so avoid extra computations when not needed. Do you spot other use cases for this function?

Reply:
not currently, but it's always nice having some helper function available when it's needed. I'll leave it up to you.  

either way, there's a typo, it should be `isSequential` and `SequentialTracks` with an `i`, and I'd suggest naming this more clearly, like `isSequentialTrackFromSameAlbum` or something similar.